rocAL/rocAL/source/coco_meta_data_reader_key_points.cpp

```diff
@@ -0,0 +1,388 @@
+        {
+            std::cout << " x : " << joints_data.joints[i][0] << " , y : " << joints_data.joints[i][1] << " , v : " << joints_data.joints_visibility[i][0] << std::endl;
+        }
+        std::cout << "Score: " <<  joints_data.score << std::endl;
+        std::cout << "Rotation: " <<  joints_data.rotation << std::endl;
+    }
+}
+
+void COCOMetaDataReaderKeyPoints::read_all(const std::string &path)
+{
+    _coco_metadata_read_time.start(); // Debug timing
+    std::ifstream f;
+    f.open(path, std::ifstream::in | std::ios::binary);
+    if (f.fail())
+        THROW("ERROR: Given annotations file not present " + path);
+    f.ignore(std::numeric_limits<std::streamsize>::max());
+    auto file_size = f.gcount();
+    f.clear(); //  Since ignore will have set eof.
+    if (file_size == 0)
+    { // If file is empty return
+        f.close();
+        THROW("ERROR: Given annotations file not valid " + path);
+    }
+    std::unique_ptr<char, std::function<void(char *)>> buff(
+        new char[file_size + 1],
+        [](char *data)
+        { delete[] data; });
+    f.seekg(0, std::ios::beg);
+    buff.get()[file_size] = '\0';
+    f.read(buff.get(), file_size);
+    f.close();
+
+    LookaheadParser parser(buff.get());
+
+    ImgSizes img_sizes;
+    JointsData joints_data;
+
+    ImgSize img_size;
+    std::vector<float> box_center, box_scale;
+    float score = 1.0;
+    float rotation = 0.0;
+    float aspect_ratio = (_out_img_width * 1.0 / _out_img_height);
+    float inverse_aspect_ratio = 1 / aspect_ratio;
+    float inverse_pixel_std = 1 / (1.0 * PIXEL_STD); 
+
+    RAPIDJSON_ASSERT(parser.PeekType() == kObjectType);
+    parser.EnterObject();
+    while (const char *key = parser.NextObjectKey())
+    {
+        if (0 == std::strcmp(key, "images"))
+        {
+            RAPIDJSON_ASSERT(parser.PeekType() == kArrayType);
+            parser.EnterArray();
+            while (parser.NextArrayValue())
+            {
+                string image_name;
+                if (parser.PeekType() != kObjectType)
+                {
+                    continue;
+                }
+                parser.EnterObject();
+                while (const char *internal_key = parser.NextObjectKey())
+                {
+                    if (0 == std::strcmp(internal_key, "width"))
+                    {
+                        img_size.w = parser.GetInt();
+                    }
+                    else if (0 == std::strcmp(internal_key, "height"))
+                    {
+                        img_size.h = parser.GetInt();
+                    }
+                    else if (0 == std::strcmp(internal_key, "file_name"))
+                    {
+                        image_name = parser.GetString();
+                    }
+                    else
+                    {
+                        parser.SkipValue();
+                    }
+                }
+                img_sizes.push_back(img_size);
```

Comment:
seems like each image has only one img_size. If so, why it is stored in a vector?

Reply:
Done